openedx/enterprise-catalog
# ENT-3547: adding caching for data from the Discovery API

## Description
* added `CatalogQueryMetadata` interface that uses cache
* only retrieve `Course` data from the Discovery API for `ContentMetadata` records that haven't been recently updated

## Ticket Link
[ENT-3547](https://openedx.atlassian.net/browse/ENT-3547)

## Testing Instructions
* Run `./manage.py update_content_metadata` from `app-shell` in devstack multiple times.
* Observe in the `worker-logs` that CatalogQuery metadata is cached on the first run and that ContentMetadata is not retrieved from the Discovery API in subsequent runs.
